Library/src/main/java/mezz/jei/library/load/registration/AdvancedRegistration.java

```diff
@@ -29,6 +36,43 @@ public void addRecipeManagerPlugin(IRecipeManagerPlugin recipeManagerPlugin) {
 		recipeManagerPlugins.add(recipeManagerPlugin);
 	}
 
+	@Override
+	public <T> void addGlobalRecipeCategoryExtension(IRecipeCategory<T> recipeCategory, IGlobalRecipeCategoryExtension<T> extension) {
+		ErrorUtil.checkNotNull(recipeCategory, "recipeCategory");
+		ErrorUtil.checkNotNull(extension, "extension");
+
+		LOGGER.info("Added global recipe category extension: {} for recipe category: {}", extension.getClass(), recipeCategory.getRecipeType().getUid());
+		recipeCategoryExtensions.put(recipeCategory, extension);
+	}
+
+	@SuppressWarnings({"rawtypes", "unchecked"})
+	@Override
+    public <T> void addGlobalRecipeCategoryExtension(Class<T> recipeClass, IGlobalRecipeCategoryExtension<T> extension) {
+        ErrorUtil.checkNotNull(recipeClass, "recipeClass");
+		ErrorUtil.checkNotNull(extension, "extension");
+
+		for (IRecipeCategory<?> recipeCategory : jeiHelpers.getRecipeCategories()) {
+			var type = recipeCategory.getRecipeType();
+			if (type.getRecipeClass().isAssignableFrom(recipeClass)) {
+				addGlobalRecipeCategoryExtension(recipeCategory, (IGlobalRecipeCategoryExtension) extension);
+			}
+		}
+    }
+
+	@SuppressWarnings({"rawtypes", "unchecked"})
+	@Override
+	public <T> void addGlobalRecipeCategoryExtension(RecipeType<T> recipeType, IGlobalRecipeCategoryExtension<T> extension) {
+		ErrorUtil.checkNotNull(recipeType, "recipeType");
+		ErrorUtil.checkNotNull(extension, "extension");
+
+		for (IRecipeCategory<?> recipeCategory : jeiHelpers.getRecipeCategories()) {
```

Comment:
instead of `jeiHelpers.getRecipeCategories()`

you can create a stream with
```java
recipeManager.createRecipeCategoryLookup()
    .limitTypes(List.of(recipeType))
    .get()
```

Reply:
And how would I grab an instance of the recipe manager at this point?

Reply:
The only option I see to do it in the way you mentioned is by storing two fields inside the advanced registration. One map for RecipeCategory -> Extensions and one for RecipeType -> Extensions and then resolve them in the RecipeGuiLogic.

Reply:
Ah ok I see the problem.
You can pass the list of recipe categories into `AdvancedRegistration` here:
https://github.com/mezz/JustEnoughItems/blob/2bbf6670b43d5d1221f286e0d13f4813f2df1a2b/Library/src/main/java/mezz/jei/library/load/PluginLoader.java#L118-L124

and avoid adding the extra access to it in the API